### ISSUE TITLE
Is there a way to programmatically pause and resume?

### ISSUE THREAD

**Issue by drastick** [metafizzy/flickity#94]
I have videos in my slides and would like to pause flickity while the video is playing and then resume it once the video is done.

**Comment by desandro** [comment]
By 'pause' do you mean disable dragging, or pause `autoPlay` ?

**Comment by drastick** [comment]
Pause autoPlay.  So while the video is playing it does not try going to the next slide but when the video is done I can programatically restart autoPlay.

**Comment by drastick** [comment]
Overall I think it would be good to provide methods for stop, play and pause.  I think I saw in another issue that you had said you didn't think it would be a good user experience to continue animating after a user clicked on a slide.  Not that I disagree with you, but I think that should be up to the person implementing Flickity and not a choice of the library itself.  This is especially true because of the commercial licensing and people paying for the library.  If someone pays to use this they should be able to chose how they use it.

**Comment by drastick** [comment]
This seems as easy as this:

```javascript
// -------------------------- Flickity -------------------------- //

// utils.extend( Flickity.defaults, {
//   autoPlay: false
// });

Flickity.createMethods.push('_createPlayer');

Flickity.prototype._createPlayer = function() {
  this.player = new Player( this );

  this.on( 'activate', this.activatePlayer );
  this.on( 'uiChange', this.stop );
  this.on( 'pointerDown', this.stop );
  this.on( 'deactivate', this.deactivatePlayer );
};

Flickity.prototype.activatePlayer = function() {
  if ( !this.options.autoPlay ) {
    return;
  }
  this.player.play();
  eventie.bind( this.element, 'mouseenter', this );
  this.isMouseenterBound = true;
};

Flickity.prototype.stop = function() {
  this.player.stop();
};

Flickity.prototype.play = function() {
  this.player.play();
};

Flickity.prototype.pause = function() {
  this.player.pause();
};

Flickity.prototype.unpause = function() {
  this.player.unpause();
};

Flickity.prototype.deactivatePlayer = function() {
  this.player.stop();
  if ( this.isMouseenterBound ) {
    eventie.unbind( this.element, 'mouseenter', this );
    delete this.isMouseenterBound;
  }
};
```

**Comment by adam-h** [comment]
You can just use `flickity_instance.activatePlayer()` and `flickity_instance.deactivatePlayer()` to play and pause respectively. Not ideal as the events have to be bound and unbound, but it works until play and pause methods are exposed.

**Comment by pesla** [comment]
+1!

**Comment by marcandrews** [comment]
I'm quite new to JS. Can someone please post a complete example of how to pause autoplay?

**Comment by drastick** [comment]
```javascript
<div class="main-gallery">...</div>

$('.main-gallery').flickity({
  // options
  cellAlign: 'left',
  contain: true
});

var flkty = Flickity.data( $('.main-gallery')[0] );

flkty.deactivatePlayer();
```

**Comment by marcandrews** [comment]
Thanks!

**Comment by jonburger** [comment]
```flkty.on('settle', function() {
	if (!flkty.player.isPlaying) {
		flkty.activatePlayer();
	}
});```

**Comment by bbnnt** [comment]
@drastick Not sure if I got it right
stuck with `Can't find variable: Player`

**Comment by desandro** [comment]
[v1.2.0](https://github.com/metafizzy/flickity/releases/tag/v1.2.0) has added [proper auto-play API](http://localhost/flickity-docs/build/api.html#player): `playPlayer`, `stopPlayer`, `pausePlayer`, `unpausePlayer`.

Closing this issue as resolved :whale:

**Comment by am05mhz** [comment]
> [v1.2.0](https://github.com/metafizzy/flickity/releases/tag/v1.2.0) has added [proper auto-play API](http://localhost/flickity-docs/build/api.html#player): `playPlayer`, `stopPlayer`, `pausePlayer`, `unpausePlayer`.
> 
> Closing this issue as resolved
 
the link is incorrect, it points to localhost, should be this https://flickity.metafizzy.co/api.html#player
